niCHARTPipelines/__main__.py

```diff
@@ -6,37 +6,281 @@
-    parser.add_argument('--DLICVmdl', type=str, help='DLICV model path.', default=None, required=True)
-    parser.add_argument('--DLMUSEmdl', type=str, help='DLMUSE Model path.', default=None, required=True)
-    parser.add_argument('--pipelineType', type=str, help='Specify type of pipeline[structural, dti, fmri].', default=None, required=True)
-    parser.add_argument('--outDir', type=str, help='Output file name with extension.', default=None, required=True)
-    parser.add_argument('--scanID', type=str, help='scan id.', default=None, required=True)
-    parser.add_argument('--derivedROIMappingsFile', type=str, help='derived MUSE ROI mappings file.', default=None, required=True)
-    parser.add_argument('--MuseROIMappingsFile', type=str, help='MUSE ROI mappings file.', default=None, required=True)
-    
-    args = parser.parse_args(sys.argv[1:])
-
-    inDir = args.inDir
-    DLICVmdl = args.DLICVmdl
-    DLMUSEmdl = args.DLMUSEmdl
-    pipelineType = args.pipelineType
-    outDir = args.outDir
+    prog = "niCHARTPipelines"
+    description = "niCHART Data Preprocessing Pipelines"
+    usage = """
+    niCHARTPipelines v{VERSION}
+    ICV calculation, brain segmentation, and ROI extraction pipelines for 
+    structural MRI data.
+
+    required arguments:
+        [INDIR]         The filepath of the directory containing the input. The 
+        [-i, --indir]   input can be a single .nii.gz (or .nii) file or a  
+                        directory containing .nii.gz files (or .nii files). 
+
+        [OUTDIR]        The filepath of the directory where the output will be
+        [-o, --outdir]  saved.
+
+        [PIPELINETYPE]  Specify type of pipeline[structural, dti, fmri]. 
+        [-p,            Currently only structural pipeline is supported.
+        --pipelinetype]
+
+        [SCANID]        Scan ID of the subject, if only one scan is given. If
+        [-s, --scanID]  multiple scans are given, this argument will be 
+                        ignored, and the scan ID will be extracted from the
+                        filename of the input file(s).
+
+        [DERIVED_ROI_MAPPINGS_FILE]     The filepath of the derived MUSE ROI 
+        [--derived_ROI_mappings_file]   mappings file.
+
+        [MUSE_ROI_MAPPINGS_FILE]    The filepath of the MUSE ROI mappings file.
+        [--MUSE_ROI_mappings_file]
+    
+    optional arguments: 
+        [DLICVMDL]      The filepath of the DLICV model will be. In case the
+        [--DLICVmdl]    model to be used is an nnUNet model, the filepath of
+                        the model's parent directory should be given. Example: 
+                        /path/to/nnUNetTrainedModels/nnUNet/
+        
+        [DLMUSEMDL]     The filepath of the DLMUSE model will be. In case the
+        [--DLMUSEmdl]   model to be used is an nnUNet model, the filepath of
+                        the model's parent directory should be given. Example:
+                        /path/to/nnUNetTrainedModels/nnUNet/
```

Comment:
Shouldn't these be removed since both models are now passed in via another param.

Reply:
I left it like that since I wanted to include the possibility of someone being able to swap one method for a different one, i.e. use nnUNet for DLICV, and something else for DLMUSE. My thinking here was to be futureproof, let's say if we train a nnUNetV2 model for one but not the other steps.